config_full.json.example

```diff
@@ -109,6 +109,11 @@
         "token": "your_telegram_token",
         "chat_id": "your_telegram_chat_id"
     },
+    "api_server": {
+        "enabled": false,
+        "listen_ip_address": "127.0.0.1",
+        "listen_port": 8080
+    },
```

Comment:
no security token? a JWT token would be great here if IP is not local.

Reply:
i was tempted to use basic auth (username, password) ... 

Adding tokens (JWT or others) means to either pre-insert a token here (which most will bother to change) - or expect unexperienced users to generate one ... - or generate one at startup and hope users see it before the page rolls over to copy/paste it to the rest-client.

Reply:
an unexperienced user who wants to control his bot by REST API by doesn't know what is security token ? :D

Reply:
JWT will require 2 calls - assuming you use the rest_client script, thats 2 calls just to get `/version` ...

Reply:
just an idea: wouldn't it better to generate a password by `secret` ? 
``` 
import secrets
secrets.token_hex() #'1476f4cfa96f20af2ca8cfdf9c5920f54d78f1b835318d723ceec2a72403cc29'
```
So leave the password empty by default and raise an error if it is empty and document how user can generate a password. basically what flask or rails does for secret session. 
(I am not insisting though but the approach seems a bit more pertinent)

Reply:
and how do you show that password to the user? 

right at startup? using docker - the password will be gone soonish (at least my configurations rotate logs after about a Meg or so)... so i'd need to restart the bot to see the password (which will probably be different each time ...) ?

Reply:
no, when you configure the bot with rest api you have to put a password yourself. that `secrets` part is just for documentation. we don't do anything but if there is no password then we raise an error.

Reply:
I am not sure which approach is better though, you can resolve this conv if you feel unnecessary.

Reply:
no it makes sense (I just understood it wrongly) :+1: